### PR TITLE
[IMP] edi_oca: Allow to use a sequence as part of exchange filename pattern

### PR DESCRIPTION
**Pull request by tuantrantg**

### CODE REVIEW

File: edi_oca/models/edi_exchange_type.py [hunk 1]
```diff
@@ -50,7 +50,25 @@ class EDIExchangeType(models.Model):
...
-    exchange_filename_pattern = fields.Char(default="{record_name}-{type.code}-{dt}")
+    exchange_filename_pattern = fields.Char(
+        default="{record_name}-{type.code}-{dt}-{seq}",
+        help="For output exchange types this should be a formatting string "
+        "with the following variables available (to be used between "
+        "brackets, `{}`): `exchange_record`, `record_name`, `type`, "
+        "`dt` and `seq`. For instance, a valid string would be "
+        "{record_name}-{type.code}-{dt}-{seq}\n"
+        "For more information:\n"
+        "- `exchange_record` means exchange reocord\n"
```
Comment (nilshamerlinck) [inline edi_oca/models/edi_exchange_type.py]:
```suggestion
        "- `exchange_record` means exchange record\n"
```

File: edi_oca/models/edi_exchange_type.py [hunk 2]
```diff
@@ -50,7 +50,25 @@ class EDIExchangeType(models.Model):
     direction = fields.Selection(
         selection=[("input", "Input"), ("output", "Output")], required=True
     )
-    exchange_filename_pattern = fields.Char(default="{record_name}-{type.code}-{dt}")
+    exchange_filename_pattern = fields.Char(
```
Comment (nilshamerlinck) [inline edi_oca/models/edi_exchange_type.py:53]: I think the help should be fixed in `edi_storage_oca` instead, can you prepare a PR on https://github.com/QuocDuong1306/edi-framework/pull/5 to change it there?

File: edi_oca/models/edi_exchange_type.py [hunk 3]
```diff
@@ -50,7 +50,25 @@ class EDIExchangeType(models.Model):
...
+        default="{record_name}-{type.code}-{dt}-{seq}",
+        help="For output exchange types this should be a formatting string "
+        "with the following variables available (to be used between "
+        "brackets, `{}`): `exchange_record`, `record_name`, `type`, "
+        "`dt` and `seq`. For instance, a valid string would be "
+        "{record_name}-{type.code}-{dt}-{seq}\n"
+        "For more information:\n"
+        "- `exchange_record` means exchange reocord\n"
+        "- `record_name` means name of the exchange reocord\n"
```
Comment (nilshamerlinck) [inline edi_oca/models/edi_exchange_type.py]:
```suggestion
        "- `record_name` means name of the exchange record\n"
```

File: edi_oca/models/edi_exchange_type.py [hunk 4]
```diff
@@ -50,7 +50,25 @@ class EDIExchangeType(models.Model):
...
+        "- `type` means code of the exchange reocord type\n"
```
Comment (nilshamerlinck) [inline edi_oca/models/edi_exchange_type.py]:
```suggestion
        "- `type` means code of the exchange record type\n"
```

File: edi_oca/models/edi_exchange_type.py [hunk 5]
```diff
@@ -50,7 +50,25 @@ class EDIExchangeType(models.Model):
...
+        "- `dt` means datetime\n"
+        "- `seq` means sequence. You can configure sequence in "
```
Comment (nilshamerlinck) [inline edi_oca/models/edi_exchange_type.py]:
```suggestion
        "- `seq` means sequence. You need a sequence to be defined in "
```

**Review by nilshamerlinck — CHANGES_REQUESTED** [review]

File: edi_oca/models/edi_exchange_type.py (hunk 2 above)
Comment (tuantrantg) [reply edi_oca/models/edi_exchange_type.py:53]: hi @nilshamerlinck I've created [a new internal PR](https://github.com/QuocDuong1306/edi-framework/pull/18) for this one

**Comment by tuantrantg** [comment]
hi @nilshamerlinck please review the PR again, thank you

**Review by nilshamerlinck — APPROVED** [review]

**Pull request closed without merging**